quickwit-core/src/indexing/split.rs

```diff
@@ -153,14 +168,23 @@ impl Split {
     }
 
     /// Merge all segments of the split into one.
-    pub async fn merge_all_segments(&mut self) -> anyhow::Result<tantivy::SegmentMeta> {
-        let segment_ids = self.index.searchable_segment_ids()?;
+    ///
+    /// If there is only one segment and it has no delete, avoid doing anything them.
+    pub async fn merge_all_segments(&mut self) -> anyhow::Result<()> {
```

Comment:
Absolutely makes sense to not return the `SegmentMeta` since we are not currently using it. :+1: